specifyweb/specify/migrations/0002_geo.py

```diff
@@ -0,0 +1,270 @@
+]
+
+def create_default_collection_types():
+    # Create default collection types for each collection, named after the discipline
+    for collection in Collection.objects.all():
+        discipline = collection.discipline
+        discipline_name = discipline.name
+        cot, created = CollectionObjectType.objects.get_or_create(
+            name=discipline_name,
+            collection=collection,
+            taxontreedef_id=discipline.taxontreedef_id
+        )
+        # Update CollectionObjects' collectionobjecttype for the discipline
+        Collectionobject.objects.filter(collection=collection).update(collectionobjecttype=cot)
+        collection.collectionobjecttype = cot
+        collection.save()
+
+def revert_default_collection_types():
+    # Reverse handeled by table deletion.
+    pass
+
+def revert_default_cog_types():
+    # Reverse handeled by table deletion
+    pass
+
+def create_default_discipline_for_tree_defs():
+    for discipline in Discipline.objects.all():
+        geography_tree_def = discipline.geographytreedef
+        geography_tree_def.discipline = discipline
+        geography_tree_def.save()
+
+        geologic_time_period_tree_def = discipline.geologictimeperiodtreedef
+        geologic_time_period_tree_def.discipline = discipline
+        geologic_time_period_tree_def.save()
+
+        lithostrat_tree_def = discipline.lithostrattreedef
+        lithostrat_tree_def.discipline = discipline
+        lithostrat_tree_def.save()
+
+        taxon_tree_def = discipline.taxontreedef
+        taxon_tree_def.discipline = discipline
+        taxon_tree_def.save()
+
+    for institution in Institution.objects.all():
+        storage_tree_def = institution.storagetreedef
+        storage_tree_def.institution = institution
+        storage_tree_def.save()
+
+def revert_default_discipline_for_tree_defs():
+    # Reverse handeled by table deletion
+    pass
+
+def create_table_schema_config_with_defaults():
+    for discipline in Discipline.objects.all():
+        for table, desc in SCHEMA_CONFIG_TABLES:
+            update_table_schema_config_with_defaults(table, discipline.id, discipline, desc)
+
+def revert_table_schema_config_with_defaults():
+    for table, _ in SCHEMA_CONFIG_TABLES:
+        revert_table_schema_config(table)
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+        ('specify', '0001_initial'),
+    ]
+    
+    def consolidated_python_django_migration_operations(apps, schema_editor):
+        create_default_collection_types()
+        create_default_discipline_for_tree_defs()
+        create_table_schema_config_with_defaults()
+
+    def renvert_cosolidated_python_django_migration_operations(apps, schema_editor):
+        revert_table_schema_config_with_defaults()
+        revert_default_collection_types()
+        revert_default_discipline_for_tree_defs()
+
```

Comment:
Yes, you are correct, this is absolutely fine. But, for the sake of understanding, swap collection_types and tree_defs (to revert mirror the forward)

there's a typo in `renvert`